weblate/static/editor/tools/search.js

```diff
@@ -0,0 +1,185 @@
+                  const sourceHighlighted = highlightMatch(
+                    userSearchInput,
+                    source.toString(),
+                  );
+
+                  // Make the URL relative
+                  const url = result.web_url.replace(/^(?:\/\/|[^/]+)*\//, "/");
+                  const resultHtml = `
+                    <a href="${url}" target="_blank" id="search-result" rel="noopener noreferrer">
+                      <small>${key}</small>
+                      <div>
+                        ${sourceHighlighted}
+                      </div>
+                    </a>
+                  `;
+
+                  // Some results are not relevant (case in-sensetive), so we filter them out
+                  if (sourceHighlighted === source) {
+                    return;
+                  }
+                  $searchPreview.append(resultHtml);
+                }
+              }
+            },
+            error: (error) => {
+              console.error("Error fetching search results: ", error);
+            },
+          });
+        }
+      }, 300); // If the user stops typing for 300ms, the search results will be fetched
+    });
+
+    // Show the preview on focus
+    $searchElment.on("focus", () => {
+      if ($searchElment.val() !== "" && $searchPreview.html() !== "") {
+        $searchPreview.show();
+      }
+    });
+
+    // Close the preview on focusout, form submit, form reset, and form clear
+    $searchElment.on("focusout", () => {
+      // Hide after a delay to allow click on a result
+      setTimeout(() => {
+        $searchPreview.hide();
+      }, 700);
+    });
+    $searchForm.on("submit", () => {
+      $searchPreview.html("");
+      $searchPreview.hide();
+    });
+    $searchForm.on("reset", () => {
+      $searchPreview.html("");
+      $searchPreview.hide();
+    });
+    $searchForm.on("clear", () => {
+      $searchPreview.html("");
+      $searchPreview.hide();
+    });
+  }
+
+  /**
+   * Builds a search query string with the user input and filters.
+   * The search query is built by concatenating the user input with the filters.
+   * The path lookup is also added to the search query.
+   * Built in the following format: `userInput path:proj/comp filters`.
+   *
+   * @param {string} searchQuery - The user input search query.
+   * @param {jQuery} $filtersElement - The jQuery object of the filters input element.
+   * @param {boolean} includeFilters - A boolean indicating whether to include the filters in the search query.
+   * @returns {string} - The built search query string.
+   *
+   * */
+  function buildSearchQuery(searchQuery, $filtersElement, includeFilters) {
+    let builtSearchQuery = searchQuery;
+
+    // Add path lookup to the search query
+    const pathSplited = window.location.pathname.split("/");
+    const projectPath = `${pathSplited[2]}/${pathSplited[3]}`;
+    if (projectPath) {
+      builtSearchQuery = `${searchQuery} path:${projectPath}`;
+    }
```

Comment:
Maybe it would be better to include this in the form instead of figuring this from the URL?

Reply:
I tried to get the `action` attr from the form but somehow I was getting `undefined` so I simply went for the pathname as it is always readily available.

Reply:
Also  the part of url manipulation is done to remove the domain and the  protocol of the `web_url`returned by the api. In local development it was returning  the ip of the server which when I go to it was unresolved (server running in wsl was port forwared to localhost).  Also not sure how it would behave in production, I didn't want to make any assumptions. So in this case a relative path would pick up the domain and the protocol of the currently loaded page. Which is I guess better.